src/domain/usecases/ListCurrentUserNotificationsUseCase.ts

```diff
@@ -1,18 +1,42 @@
-import { NotificationRepository } from "../repositories/NotificationRepository";
+import _ from "lodash";
+import { NotificationListOptions, NotificationRepository } from "../repositories/NotificationRepository";
 import { FutureData } from "../types/Future";
 import { Notification, NotificationWildcard } from "../entities/Notification";
 import { UserRepository } from "../repositories/UserRepository";
+import { User } from "../entities/User";
 
 export class ListCurrentUserNotificationsUseCase {
     constructor(private notificationRepository: NotificationRepository, private userRepository: UserRepository) {}
 
     public execute(): FutureData<Notification[]> {
-        return this.userRepository.getCurrentUser().flatMap(user => {
-            return this.notificationRepository.list({
-                user: user,
-                isRead: false,
-                wildcard: [NotificationWildcard.ALL, NotificationWildcard.WEB],
-            });
-        });
+        const notificationFilter: NotificationListOptions = {
+            wildcard: [NotificationWildcard.ALL, NotificationWildcard.WEB, NotificationWildcard.Both],
+        };
+        return this.userRepository
+            .getCurrentUser()
+            .flatMap(user =>
+                this.notificationRepository.list(notificationFilter).map(notifications => ({ user, notifications }))
+            )
+            .map(({ user, notifications }) => this.filterUserNotifications(notifications, user));
```

Comment:
[no need to change] this.filterUserNotifications can be directly called in the flatMap block, but this is also valid.